src/Websites/Websites/Cmdlets/TrafficRouting/RemoveAzureWebAppTrafficRoutingRule.cs

```diff
@@ -0,0 +1,80 @@
+
+namespace Microsoft.Azure.Commands.WebApps.Cmdlets.TrafficRouting
+{
+    /// <summary>
+    /// this commandlet will let you Remove the given Azure App Service Traffic Routing using ARM APIs
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, ResourceManager.Common.AzureRMConstants.AzurePrefix + "WebAppTrafficRouting", DefaultParameterSetName = RoutingParameterSet, SupportsShouldProcess = true), OutputType(typeof(RampUpRule))]
+    public class RemoveAzureWebAppTrafficRoutingRuleCmdlet : WebAppBaseClientCmdLet
+    {
+        private const string RoutingParameterSet = "RoutingParameterSet";
+
+        [Parameter(Mandatory = true, ParameterSetName = RoutingParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = RoutingParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string WebAppName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = RoutingParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string RuleName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Return the Routing Rule object.")]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (!string.IsNullOrWhiteSpace(ResourceGroupName) && !string.IsNullOrWhiteSpace(WebAppName))
+            {
+                var webApp = new PSSite(WebsitesClient.GetWebApp(ResourceGroupName, WebAppName, null));
+                SiteConfig siteConfig = webApp.SiteConfig;
+
+                //Check for Rule existance with the given name.
+                var givenRampUpRuleObj = siteConfig.Experiments.RampUpRules.FirstOrDefault(item => item.Name == RuleName);
+                if (givenRampUpRuleObj != null)
+                {
+                    if (this.ShouldProcess(this.RuleName, string.Format("Deleting Routing Rule for slot - '{0}' from Web Application - {1}", this.RuleName, this.WebAppName)))
+                    {
+                        //Remove the rule
+                        siteConfig.Experiments.RampUpRules.Remove(givenRampUpRuleObj);
+                        // Update web app configuration
+                        WebsitesClient.UpdateWebAppConfiguration(ResourceGroupName, webApp.Location, WebAppName, null, siteConfig, null, null, null);
+                        if (PassThru.IsPresent)
+                        {
+                            var app = WebsitesClient.GetWebApp(ResourceGroupName, WebAppName, null);
+                            WriteObject(app.SiteConfig.Experiments.RampUpRules.FirstOrDefault(rule => rule.Name == RuleName));
+                        }
+                    }
+                }
+                else
+                {
+                    throw new ValidationMetadataException(string.Format("Given Routing Rule with name '{0}' in WebApp '{1}' is not present." +
```

Comment:
Please use resource strings for customer-facing messages.

Reply:
It also avoids duplicated hardcoded string

Reply:
> Please use resource strings for customer-facing messages.

Not clear on resource strings. Can you give me more details on it.

Reply:
If you go to `Websites/Properties/Resoures.resx` in visual studio you'll find a dictionary of strings, simply add you string there and reference it by `Resources.{your-key}` in code.

Resource strings is a way to centralize string resources in your project. It's helpful for reducing code duplication and for i18n as well.